src/OrchardCore/OrchardCore.Localization.Core/DataAnnotations/LocalizedDataAnnotationsMvcOptions.cs

```diff
@@ -0,0 +1,28 @@
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Localization;
+using Microsoft.Extensions.Options;
+
+namespace OrchardCore.Localization.DataAnnotations
+{
+    internal class LocalizedDataAnnotationsMvcOptions : IConfigureOptions<MvcOptions>
+    {
+        private readonly IServiceScopeFactory _serviceScopeFactory;
+
+        public LocalizedDataAnnotationsMvcOptions(IServiceScopeFactory serviceScopeFactory)
+        {
+            _serviceScopeFactory = serviceScopeFactory;
+        }
+
+        public void Configure(MvcOptions options)
+        {
+            using (var scope = _serviceScopeFactory.CreateScope())
+            {
+                var serviceProvider = scope.ServiceProvider;
+                var localizer = serviceProvider.GetService<IStringLocalizerFactory>().Create(typeof(LocalizedDataAnnotationsMvcOptions));
+
```

Comment:
When executed in a tenant context there is already an ambient scope service provider that you can access through an `IHttpContextAccessor` that you could inject in the ctor, or directly by using `ShellScope.Services`.

But but, because `IStringLocalizerFactory` seems to be a tenant singleton, you can inject it directly in the ctor and then use it in the `Configure()` method without having to create a scope.

Reply:
Let me try `ShellScope.Services`

> But but, because IStringLocalizerFactory seems to be a tenant singleton, you can inject it directly in the ctor and then use it in the Configure() method without having to create a scope.

Which `Configure` method do you mean?

Reply:
This one.

I meant inject `IStringLocalizerFactory` directly in your ctor

And then use it in your `public void Configure(MvcOptions options)`

Reply:
Got it, let me try

Reply:
I update the PR, I'm not sure why I didn't inject `IStringLocalizerFactory` from the beginning ;)